rosbag2_transport/src/rosbag2_transport/qos.cpp

```diff
@@ -21,6 +21,53 @@
 
 namespace
 {
+
+// Foxy compatibility - copied from Galactic version of rmw/time.c
+// Introuced in https://github.com/ros2/rmw/pull/301
+
+typedef rcutils_duration_value_t rmw_duration_t;
+// we can't use the Galactic version of this constant because Foxy RMWs won't understand it
+static const rmw_time_t RMW_DURATION_INFINITE {0LL, 0LL};
+
+
+rmw_duration_t
+rmw_time_total_nsec(const rmw_time_t time)
+{
+  static const uint64_t max_sec = INT64_MAX / RCUTILS_S_TO_NS(1);
```

Comment:
this is an implicit cast from int64_t to uint64_t. I'd suggest explicitly casting or using auto.

Reply:
Thanks for the feedback, but this is copy-pasted from the C code we're calling in Galactic, I won't be messing with it here.